Paternity Test.py

```diff
@@ -0,0 +1,378 @@
+get_ipython().system('gsutil ls $directory > contentlocations.txt')
+logging.info("Processing filenames...")
+get_ipython().system("cat contentlocations.txt | sed 's@.*/@@' > filenames.txt")
+
+# Import everything
+logging.info("Constructing dataframe...")
+data={}
+this_file=open("contentlocations.txt", "r")
+lineslocation = this_file.read().splitlines()
+this_file.close()
+that_file=open("filenames.txt", "r")
+linesfilename = that_file.read().splitlines()
+data['filename'] = linesfilename
+data['location'] = lineslocation #here in order to put columns in a particular order without reassigning later
+
+# Create dataframe
+df = pd.DataFrame(data)
+
+# Split on file extension
+df['FileType'] = df.filename.str.rsplit('.', 1).str[-1]
+
+# Count file extensions to see how many child file types there are
+unique_children = pd.unique(df[['FileType']].values.ravel("K"))
+unique_children = unique_children[unique_children!=PARENT_FILETYPE]
+
+# Play with strings to get basic ID, needed to link parents and children
+logging.info("Manipulating dataframe...")
+i = 0
+for child_extension in unique_children: #loop once per child extension
+    i=i+1
+    logging.debug("\t* Dealing with %s (child %d out of %d)" % (child_extension, i, len(unique_children)))
+    df[child_extension] = ""
+    df[child_extension+"_location"] = ""
+    for index_label, row_series in df.iterrows():
+        # On first iteration, iterate over parent files
+        # Only do once to avoid wasting time repeating this constantly
+        if (df.at[index_label,'FileType'] == PARENT_FILETYPE) & (i==1):
+            parent_baseID = baseID(row_series['filename'], child_extension)
+            df.at[index_label,'ID'] = parent_baseID
+        # Only iterate over children that match the child_extension
+        # This avoids overwriting if there's more than one child_extension
+        elif df.at[index_label,'FileType'] == child_extension:
+            child_baseID = baseID(row_series['filename'], child_extension)
+            df.at[index_label,'ID'] = child_baseID
+            #output_list.update({baseID(row_series['filename'], child_extension) : df.at[index_label,'FileType']})
+        else:
+            pass
+# For thousands of files this takes a long time, let's try to show some sort of progress.
+# Progress bar packages that use carriage returns don't play nicely with Jupyter when cells
+# are ran more than once, so a balance between the user being updated and not making several
+# lines of text is needed.
+if(df.shape[0] > 100):
+    triggerevery = df.shape[0] // 5
+    nexttrigger = triggerevery
+else:
+    triggerevery = None
+
+logging.info("Child file types are: %s  If this looks wrong, check your GCS directory for missing or junk files." % unique_children)
+
+# Iterate yet again to match children with their parents
+logging.info("Matching children and parents...")
+i = 0 #only used for logging
+j = 0
+for child_extension in unique_children: #loop once per child extension
+    i=i+1
```

Comment:
You may want to look at `enumerate`: https://book.pythontips.com/en/latest/enumerate.html